src/jvmMain/kotlin/fr/acinq/lightning/io/JvmTcpSocket.kt

```diff
@@ -58,34 +64,91 @@ class JvmTcpSocket(val socket: Socket) : TcpSocket {
         socket.close()
     }
 
+
+    companion object {
+        private val logger by lightningLogger<JvmTcpSocket>()
+        fun buildPublicKey(encodedKey: ByteArray): java.security.PublicKey {
+            val spec = X509EncodedKeySpec(encodedKey)
+            val algorithms = listOf("RSA", "EC")
```

Comment:
@sstone we only support `RSA` and `EC` to match what's done in Swift (see https://github.com/ACINQ/lightning-kmp/blob/9b019b16f7e6c3fab69a1ce122b2d0061e842d5e/PhoenixCrypto/PhoenixCrypto/Classes/NativeSocket.swift#L180-L191).

We could add more though (see https://docs.oracle.com/en/java/javase/11/docs/specs/security/standard-names.html#keyfactory-algorithms)

Reply:
On iOS, the documentation is quite sparse. For the `kSecAttrKeyClass` value, those are the only 3 options I could find. It might be `kSecAttrKeyTypeEC` on iOS is equivalent to `[EC, XDH, X25519, X448]` on Android. :man_shrugging: I have no clue. But I think it's safe if you add all of them.

Reply:
Done in b5eebc2dacdd4ff37ca13b46c57693adccd72322